src/main/java/seedu/address/logic/parser/BuyCommandParser.java

```diff
@@ -20,21 +26,32 @@
  * Parses input arguments and creates a new {@code BuyCommand} object
  */
 public class BuyCommandParser implements Parser<BuyCommand> {
+
+    private final DateTimeFormatter oldPattern = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private final DateTimeFormatter newPattern = DateTimeFormatter.ofPattern("dd/MM/yyyy");
```

Comment:
A suggestion would to be to delare these variables as public static final in a utility class since they are redeclared a couple of times in the files.